modules/modelSetup/PixArtAlphaLoRASetup.py

```diff
@@ -65,12 +65,12 @@ def setup_model(
     ):
         if model.text_encoder_lora is None and config.train_text_encoder:
             model.text_encoder_lora = LoRAModuleWrapper(
-                model.text_encoder, config.lora_rank, "lora_te", config.lora_alpha
+                model.text_encoder, config.lora_rank, "lora_te", config.lora_alpha, config.dropout_pct
```

Comment:
Currently, the LoRAmoduleWrapper is created in two places. The Setup class and the ModelLoader class. I would like to refactor that at some point. But at the moment, the dropout value still needs to be passed in both places, not just here. If you only pass it here, it won't work when loading an existing LoRA

Reply:
ModelLoader requires a bit of thought to do this cleanly, so I'd like to know what you think before I push any changes. We've got a couple of options here:

(1) Pass the training config to the ModelLoader constructor, save the config. Now anything that inherits or mixes ModelLoader can access this. Possible problem: does the ModelLoader object get recreated when a new TrainConfig is made (like when a new training is started)? If not, this won't work.

(2) Pass the training config to the ModelLoader.load() method. More places will need to pass it and possibly store it, but no concerns over the ModelLoader/TrainConfig lifetimes.

Reply:
Passing the entire config is not a good idea. That would only work when training. If you want to convert a model, you don't have access to a TrainConfig class. 

You could add a function to the LoRAModelWrapper that changes the dropout after it's created. Then call that function from the setup class

Reply:
Think there might have been a bit of a misunderstanding here. I understand that it would only work when training. The idea is to pass to the ModelLoader an optional TrainConfig, so it can pull the dropout probability from it when it exists. It gets ignored for places that don't pass the TrainConfig.

If I do that, the question around lifetimes of TrainConfig and when to provide the TrainConfig to the ModuleLoader remains. It can be given as part of the load() method, or as part of the constructor. The constructor requires substantially less threading it through method parameters, but the question of variable lifetime remains (e.g., if ModuleLoader is only created once, and future updates to the config in the UI and pressing the "Start Training" button doesn't re-create it, that would be a problem with providing the TrainConfig to the constructor).

Reply:
I understand you question, but I think the ModelLoader instance should be completely independent from the TrainConfig. Even if it's just an optional parameter.

Instead of passing the dropout to the LoRAModuleWrapper constructor, you can set it in the setup class after the lora was loaded. This way you don't need to add it in two places. For example, the setup_model function could look like this:
```python
        if model.text_encoder_lora is None:
            model.text_encoder_lora = LoRAModuleWrapper(
                model.text_encoder, config.lora_rank, "lora_te", config.lora_alpha
            )
        model.text_encoder_lora.set_dropout(config.dropout_probability)
```

In the constructor, you can always use a default value of 0